types/vote.go

```diff
@@ -272,38 +224,11 @@ func (vote *Vote) ValidateBasic() error {
 		return fmt.Errorf("signature is too big (max: %d)", MaxSignatureSize)
 	}
 
-	// XXX: add length verification for vote extension?
-
-	return nil
-}
-
-func (ext VoteExtension) Copy() VoteExtension {
-	res := VoteExtension{
-		AppDataToSign:             ext.AppDataToSign,
-		AppDataSelfAuthenticating: ext.AppDataSelfAuthenticating,
+	if len(vote.Extension) > 0 && len(vote.ExtensionSignature) == 0 {
```

Comment:
The extension signature should _always_ be there, even for empty extensions.
Is this a temporary workaround for the e2e tests? If yes, please add a TODO. If not, please remove the first part of the condition

Reply:
TODO added :slightly_smiling_face: